src/image.rs

```diff
@@ -0,0 +1,275 @@
+use std::{collections::HashMap, env::var, thread::sleep, time::Duration};
+use testcontainers::{
+    core::{Container, Docker, Port, WaitForMessage},
+    Image,
+};
+
+// TODO: Switch to using 18081 port since --regtest is mainnet
```

Comment:
I disagree here. `regtest` should use a different port than mainnet. While `regtest` uses `mainnet` settings `regtest` != `mainnet`. Imagine you run 2 nodes: `regtest` (for testing) and mainnet (for production).

Reply:
Posting here for continuity, we chatted in a call. Going to use 48081 (since currently there is significance to the following port numbers within the monero community

18081 - mainnet
28081 - stagenet
38081 - testnet

Reply:
Does monero run on different ports per network by default?

This is being run as a docker container through testcontainers, so the port that is exposed on the host machine is some random one anyway and you have to fetch it with `get_mapped_port`.

More generally: While I get the argument, I am not convinced that it actually adds value. On the contrary, writing software against an Ethereum node (which always uses `8545`) is so much easier than for Bitcoin. Just have a look at our configuration file handling and all the extra code we have to write just to get the port settings correct.

If "I want to run two different nodes on my machine" is the only argument _for_ this behaviour, then I don't think it was a good trade-off by the Bitcoin team to implement this behaviour. The default case is to only run a single instance. I should not need to write a lot of code to handle the default case. If someone is running two at the same time, they should have to do some work to make that happen.